# Drawing Conclusions
Use the space below to address questions on datasets `clean_08.csv` and `clean_18.csv`. You should've created these data files in the previous section: *Fixing Data Types Pt 3*.

In [1]:
# load datasets
import pandas as pd
df_08 = pd.read_csv("clean_08.csv")
df_18 = pd.read_csv("clean_18.csv")

### Q1: Are more unique models using alternative sources of fuel? By how much?

In [7]:
df_08.shape

(987, 13)

In [8]:
df_08.head()

model  displ  cyl    trans drive      fuel    veh_class  \
0  ACURA MDX    3.7    6  Auto-S5   4WD  Gasoline          SUV   
1  ACURA RDX    2.3    4  Auto-S5   4WD  Gasoline          SUV   
2   ACURA RL    3.5    6  Auto-S5   4WD  Gasoline  midsize car   
3   ACURA TL    3.2    6  Auto-S5   2WD  Gasoline  midsize car   
4   ACURA TL    3.5    6  Auto-S5   2WD  Gasoline  midsize car   

   air_pollution_score  city_mpg  hwy_mpg  cmb_mpg  greenhouse_gas_score  \
0                  7.0        15       20       17                     4   
1                  7.0        17       22       19                     5   
2                  7.0        16       24       19                     5   
3                  7.0        18       26       21                     6   
4                  7.0        17       26       20                     6   

  smartway  
0       no  
1       no  
2       no  
3      yes  
4      yes

In [4]:
df_08['fuel'].value_counts()

Gasoline    984
ethanol       1
CNG           1
gas           1
Name: fuel, dtype: int64

In [9]:
df_18.shape

(832, 13)

In [5]:
df_18['fuel'].value_counts()

Gasoline       749
Gas             26
Ethanol         26
Diesel          19
Electricity     12
Name: fuel, dtype: int64

In [12]:
print(3*100/987)
print((26+26+19+12))
print((26+26+19+12)*100/832)

0.303951367781155
83
9.975961538461538


More unique models are using alternative sources of fuel now
In 2008 it was 3 models, now it's 83
That's an increase from 0.3% of all models to 9.98%

### Q2: How much have vehicle classes improved in fuel economy?  

In [19]:
df_08_cmb = df_08.groupby('veh_class')['cmb_mpg'].mean()

In [20]:
df_18_cmb = df_18.groupby('veh_class')['cmb_mpg'].mean()

In [22]:
df_08_cmb

veh_class
SUV              18.471429
large car        18.509091
midsize car      21.601449
minivan          19.117647
pickup           16.277108
small car        21.105105
station wagon    22.366667
van              14.952381
Name: cmb_mpg, dtype: float64

In [23]:
df_18_cmb

veh_class
large car          23.409091
midsize car        27.884058
minivan            20.800000
pickup             18.589744
small SUV          24.074074
small car          25.421053
special purpose    18.500000
standard SUV       18.197674
station wagon      27.529412
Name: cmb_mpg, dtype: float64

In [21]:
(df_18_cmb-df_08_cmb)/df_08_cmb

veh_class
SUV                     NaN
large car          0.264735
midsize car        0.290842
minivan            0.088000
pickup             0.142079
small SUV               NaN
small car          0.204498
special purpose         NaN
standard SUV            NaN
station wagon      0.230823
van                     NaN
Name: cmb_mpg, dtype: float64

### Q3: What are the characteristics of SmartWay vehicles? Have they changed over time?

In [27]:
df_08.groupby('smartway').mean()

displ       cyl  air_pollution_score   city_mpg    hwy_mpg  \
smartway                                                                  
no        4.056837  6.771005             6.293245  15.133443  21.299835   
yes       2.602895  4.826316             7.365789  20.984211  28.413158   

            cmb_mpg  greenhouse_gas_score  
smartway                                   
no        17.316310              4.149918  
yes       23.736842              6.868421

In [28]:
df_18.groupby('smartway').mean()

displ       cyl  air_pollution_score   city_mpg    hwy_mpg  \
smartway                                                                  
Elite     1.970588  4.000000             7.000000  53.235294  53.117647   
No        3.270304  5.720994             3.798343  19.325967  26.708564   
Yes       1.753846  3.923077             4.879121  31.483516  39.296703   

            cmb_mpg  greenhouse_gas_score  
smartway                                   
Elite     53.000000             10.000000  
No        22.041436              4.276243  
Yes       34.439560              7.538462

### Q4: What features are associated with better fuel economy?

In [29]:
df_08['cmb_mpg'].mean()

19.78824721377913

In [31]:
df_08['good_fuel_econ'] = df_08['cmb_mpg'] > df_08['cmb_mpg'].mean()

In [32]:
df_08.groupby('good_fuel_econ').mean()

displ       cyl  air_pollution_score   city_mpg    hwy_mpg  \
good_fuel_econ                                                                  
False           4.416667  7.277778             6.382479  14.134615  20.085470   
True            2.667823  4.890173             6.998073  20.317919  27.603083   

                  cmb_mpg  greenhouse_gas_score  
good_fuel_econ                                   
False           16.235043              3.596154  
True            22.992293              6.639692